railties/lib/rails/tasks/framework.rake

```diff
@@ -66,3 +66,23 @@ namespace :rails do
     end
   end
 end
+
+namespace :rails do
+  task :deprecated do
+    Rake.application.top_level_tasks.grep(/^rails:/).each do |task|
+      $stderr.puts "WARNING: #{task} is deprecated. Please use #{task.sub(/^rails:/, 'app:')} instead."
+    end
```

Comment:
Don't need to print deprecation warnings for every task in the namespace. Likely more confusing than just printing the task they ran, if we can.

The message should mention that it's the namespace that is deprecated. Going with something like:

```Running this task with the rails task namespace is deprecated in favor of app. Please run e.g. bin/rails app:update instead.
```

Could even remove the need to print the task altogether.

We should also require `active_support` and use `ActiveSupport::Deprecation.warn`, so people get their configured deprecation behavior to fire.